File: aoc_2024/day_22/p1.py
from utils.file import read_input

contents = read_input(__file__)

lines = map(int, contents.splitlines())


def secret(n: int) -> int:
    n ^= n * 64
    n %= 16777216
    n ^= n // 32
    n %= 16777216
    n ^= n * 2048
    n %= 16777216
    return n


total = 0
for n in lines:
    for i in range(2000):
        n = secret(n)
    total += n

print(total)

File: aoc_2024/day_1/p1.go
package main

import (
	"aoc/utils"
	"slices"
	"strings"
)

func main() {
	lines := utils.MustReadLines("input")
	left, right := []int{}, []int{}
	for line := range lines {
		parts := strings.Split(line, "   ")
		a, b := utils.MustAtoi(parts[0]), utils.MustAtoi(parts[1])
		left, right = append(left, a), append(right, b)
	}
	slices.Sort(left)
	slices.Sort(right)
	total := 0
	for i, leftNum := range left {
		rightNum := right[i]
		total += utils.Abs(rightNum - leftNum)
	}
	println("total", total)
}
